Logging/src/V2/Gapic/ConfigServiceV2GapicClient.php

```diff
@@ -771,244 +1388,784 @@ public function getSink($sinkName, array $optionalArgs = [])
+     * }
+     *
+     * @return \Google\Cloud\Logging\V2\LogBucket
+     *
+     * @throws ApiException if the remote call fails
+     */
+    public function getBucket($name, array $optionalArgs = [])
+    {
+        $request = new GetBucketRequest();
+        $requestParamHeaders = [];
+        $request->setName($name);
+        $requestParamHeaders['name'] = $name;
+        $requestParams = new RequestParamsHeaderDescriptor($requestParamHeaders);
+        $optionalArgs['headers'] = isset($optionalArgs['headers']) ? array_merge($requestParams->getHeader(), $optionalArgs['headers']) : $requestParams->getHeader();
+        return $this->startCall('GetBucket', LogBucket::class, $optionalArgs, $request)->wait();
+    }
+
+    /**
+     * Gets the Logs Router CMEK settings for the given resource.
+     *
+     * Note: CMEK for the Logs Router can currently only be configured for GCP
+     * organizations. Once configured, it applies to all projects and folders in
+     * the GCP organization.
+     *
+     * See [Enabling CMEK for Logs
+     * Router](https://cloud.google.com/logging/docs/routing/managed-encryption)
+     * for more information.
+     *
+     * Sample code:
+     * ```
+     * $configServiceV2Client = new ConfigServiceV2Client();
+     * try {
+     *     $formattedName = $configServiceV2Client->cmekSettingsName('[PROJECT]');
+     *     $response = $configServiceV2Client->getCmekSettings($formattedName);
+     * } finally {
+     *     $configServiceV2Client->close();
+     * }
+     * ```
+     *
+     * @param string $name         Required. The resource for which to retrieve CMEK settings.
+     *
+     *                             "projects/[PROJECT_ID]/cmekSettings"
+     *                             "organizations/[ORGANIZATION_ID]/cmekSettings"
+     *                             "billingAccounts/[BILLING_ACCOUNT_ID]/cmekSettings"
+     *                             "folders/[FOLDER_ID]/cmekSettings"
+     *
+     *                             Example: `"organizations/12345/cmekSettings"`.
+     *
+     *                             Note: CMEK for the Logs Router can currently only be configured for GCP
+     *                             organizations. Once configured, it applies to all projects and folders in
+     *                             the GCP organization.
+     * @param array  $optionalArgs {
+     *     Optional.
+     *
+     *     @type RetrySettings|array $retrySettings
+     *           Retry settings to use for this call. Can be a
+     *           {@see Google\ApiCore\RetrySettings} object, or an associative array of retry
+     *           settings parameters. See the documentation on
+     *           {@see Google\ApiCore\RetrySettings} for example usage.
+     * }
+     *
+     * @return \Google\Cloud\Logging\V2\CmekSettings
+     *
+     * @throws ApiException if the remote call fails
+     */
+    public function getCmekSettings($name, array $optionalArgs = [])
```

Comment:
This new method signature breaks BC. `$name` moved from an optional argument to a required argument.  The previous method signature looked like this:

```php
public function getCmekSettings(array $optionalArgs = [])
```

This shouldn't be allowed by the API, so we need to look into how this happened.